### PR TITLE
[WIP] Exploratory POC for Upgrade to Docusaurus v2

### PR DESCRIPTION
**Pull request by danforbes**
Re #546 & https://github.com/facebook/docusaurus/issues/776

### CODE REVIEW

**Comment by slorber** [comment]
Hi,

Before all, just wanted to say that D2 does not support yet translations, cf https://github.com/slorber/substrate-developer-hub.github.io/commit/f3fe79b14bebe60e08dc7dcffb54dd7c0025651c

File: website/src/pages/index.js [hunk 1]
```diff
@@ -16,14 +16,13 @@
...
-const Col = require('../../../../react-bootstrap/Col')
-const Image = require('../../../../react-bootstrap/Image')
-const translate = require('../../server/translate').translate
+const Container = require('react-bootstrap/Container')
+const Button = require('react-bootstrap/Button')
+const Row = require('react-bootstrap/Row')
+const Col = require('react-bootstrap/Col')
+const Image = require('react-bootstrap/Image')
 
 class Index extends React.Component {
```
Comment (slorber) [inline website/src/pages/index.js:27]: you should use export default class Index

File: website/src/pages/index.js [hunk 2]
```diff
@@ -50,7 +49,7 @@ class Index extends React.Component {
 					}
 					buttonText={<translate>Get Started</translate>}
 					buttonUrl={`tutorials`}
-				/>
+				/> */}
 
 				<section className='mainContainer' id='home'>
```
Comment (slorber) [inline website/src/pages/index.js:54]: this is not valid JSX? you should use double quotes, don't know why you have single quotes everywhere, didn't even know it was possible before :o

Comment (slorber) [reply website/src/pages/index.js:54]: oh, actually seems to work :o

File: website/src/pages/index.js [hunk 3]
```diff
@@ -16,14 +16,13 @@
...
-const Button = require('../../../../react-bootstrap/Button')
-const Row = require('../../../../react-bootstrap/Row')
-const Col = require('../../../../react-bootstrap/Col')
-const Image = require('../../../../react-bootstrap/Image')
-const translate = require('../../server/translate').translate
+const Container = require('react-bootstrap/Container')
+const Button = require('react-bootstrap/Button')
+const Row = require('react-bootstrap/Row')
+const Col = require('react-bootstrap/Col')
+const Image = require('react-bootstrap/Image')
```
Comment (slorber) [inline website/src/pages/index.js:25]: you should rather use ES6 imports/exports

or you'd have to use

const Container = require('react-bootstrap/Container').default

File: website/package.json [hunk 4]
```diff
@@ -1,23 +1,28 @@
...
     "crowdin-upload": "crowdin --config ../crowdin.yaml upload sources --auto-update -b master",
     "crowdin-download": "crowdin --config ../crowdin.yaml download -b master",
     "check-links": "yarn blc http://localhost:3000 -ro --exclude \"http://localhost:3000/docs/ru\" --exclude \"http://localhost:3000/docs/zh-CN\" --exclude \"https://playground.substrate.dev/\" --exclude \"https://player.twitch.tv/?channel=paritylivecoding\" --exclude \"http://localhost:8000/\""
   },
   "devDependencies": {
-    "broken-link-checker": "^0.7.8",
-    "docusaurus": "^1.14.4"
+    "@docusaurus/core": "^2.0.0-alpha.40",
+    "@docusaurus/preset-classic": "^2.0.0-alpha.40",
+    "@docusaurus/theme-classic": "^2.0.0-alpha.37",
```
Comment (slorber) [inline website/package.json:17]: newer versions highly recommended, "alpha.56" or upper

Comment (slorber) [reply website/package.json:17]: This seems to fix the docs pages. At least I'm able to see something without errors even if it's far from perfect:

image

File: website/src/pages/index.js [hunk 5]
```diff
@@ -36,7 +35,7 @@ class Index extends React.Component {
 
 		return (
 			<section>
```
Comment (slorber) [inline website/src/pages/index.js:37]: you need to apply the theme/Layout component around all your pages, otherwise they won't have any layout

**Comment by slorber** [comment]
https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/pull/568

This PR makes it possible to display homepage + docs, even if it's far from perfect at least you have something displayed now

but you'd likely have to rewrite all your imports, all your siteConfig usage, all your <translate> tags... I'd recommend also using functional comps + hooks instead of classes

maybe it's a bit early for a migration if you need i18n

**Comment by danforbes** [comment]
Thank you for your contributions, all. I am closing this exploratory PR.

**Pull request closed without merging**